### PR TITLE
updated environment.yaml

### PR DESCRIPTION
**Pull request by cunningjames**
This brings the environment.yaml file up-to-date. I believe all packages are at their latest conda-available versions, and the Python version has been set to 3.11. I get the following warning when generating the Conda environment, which I find a little bizarre because it doesn't seem to reflect the contents of the file:

> Collecting package metadata (repodata.json): \ WARNING conda.models.version:get_matcher(546): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.9.0.*, but conda is ignoring the .* and treating it as 1.9.0
> WARNING conda.models.version:get_matcher(546): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.8.0.*, but conda is ignoring the .* and treating it as 1.8.0

But it appears to be harmless.

I have generated and looked over the slides using this environment, and largely everything appears to work. There are no obvious errors. There are, however, a few mlflow warnings which don't seem to impact the content otherwise ... but I'll try to fix them this weekend just the same.

### CODE REVIEW

**Comment by augustopher** [comment]
I switched this over to develop for the base branch, which luckily pulled some case study updates it seems we were lacking. I'm going to test it locally real quick, but I think we're probably good for approval. The CI failure are from sklearn syntax updates I already fixed on develop, so we can check the CI after merge and fix any lingering issues.

**Review by augustopher — APPROVED** [review]

**Review by augustopher — COMMENTED** [review]
Based on the CI error, and trying to replicate locally... in the 09 notebook, when running the first example under "Hyperparameter Tuning" (cell run 10), it seems that the `df = mlflow.search_runs(experiment_ids='1')` in cell 11 is returning an empty data frame, which leads to no tag columns, hence the key error when running `df['tags.mlflow.runName']` in cell 12. Not sure if this is a syntax update issue with `search_runs`, or if the tuning runs need changes to produce the needed results.

But this is all taking a very long time to run in CI and locally, and editing the grid or CV folds to reduce run time seems to be masking the potential error in the results df.

**Comment by augustopher** [comment]
Exploring `mlflow.search_runs` without a filter, it seems that the `experiment_id` values need updating to find the right results. The docs say using `None` (the default) will give the active experiment, or we can use `experiment_names` instead.

**Comment by augustopher** [comment]
After resetting my mlflow runs and starting fresh, simply removing the `experiment_ids` filter worked just fine. The only caveat is that having multiple runs (namely re-runs and failures), the code for getting the ID will fail, since it assumes only one result after filtering. So we might want to filter to the first/newest run, or just advise restarting the kernel and deleting mlflow files before re-running if an error occurs.

**Comment by cunningjames** [comment]
@augustopher 

> Not sure if this is a syntax update issue with search_runs, or if the tuning runs need changes to produce the needed results.

For what it's worth, I can't replicate any such issues locally. I was referring to warnings that MLflow has truncated a key that it logs, seemingly because Scikit-Learn's `__repr__`s are too long.

**Comment by cunningjames** [comment]
Scratch that. My environment was borked.
